packages/curve-ui-kit/src/hooks/useLocalStorage.ts

```diff
@@ -25,14 +25,20 @@ function getStoredValue<Type, Default>(key: string, initialValue: Default | unde
  */
 export function useLocalStorage<Type, Default = Type>(key: string, initialValue?: Default): GetAndSet<Type, Default> {
   type T = Type | Default
-  const value = getStoredValue<Type, Default>(key, initialValue)
+  const storageItem = window.localStorage.getItem(key)
+  const storedValue = useMemo(() => (storageItem == null ? null : (JSON.parse(storageItem) as T)), [storageItem])
```

Comment:
```suggestion
  const storedValue = useMemo(() => getFromLocalStorage<T>(key), [key])
```

Reply:
no, this doesn't work, the other useMemo's won't detect when the storage value changes